### PR TITLE
test: ++ SimpleVault to demo system contract mocks

### PR DESCRIPTION
**Pull request by mshakeg**
**Description**:

Demos how to utilize the Hedera system contract mocks in a hardhat project(against a local hardhat node) by duplicating an instance of a given mock contract to the Hedera defined system contract address using the `hardhat_setCode` method.

**Related issue(s)**:

Fixes the hardhat component of https://github.com/hashgraph/hedera-json-rpc-relay/issues/1172

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by Nana-EC** [comment]
@mshakeg please rebase, we'll review this sprint

**Comment by Nana-EC** [comment]
@AlfredoG87 & @ebadiere please review

File: tools/hardhat-example/.env.example [hunk 1]
```diff
@@ -2,3 +2,5 @@
 OPERATOR_PRIVATE_KEY=0x105d050185ccb907fba04dd92d8de9e32c18305e097ab41dadda21489a211524
 RECEIVER_PRIVATE_KEY=0x2e1d968b041d84dd120a5860cee60cd83f9374ef527ca86996317ada3d0d03e7
 RELAY_ENDPOINT='http://localhost:7546'
+
+MNEMONIC="here is where your twelve words mnemonic should be put my friend"
```
Comment (AlfredoG87) [inline tools/hardhat-example/.env.example:6]: please add a blank line at the end of the file

File: tools/hardhat-example/test/simple-vault.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        second: 0,
+        autoRenewAccount: deployerAddress,
+        autoRenewPeriod: 0
+      }
+    };
+
+    // - - - - - DEPLOY HTS Token via direct EOA call to HTS System Contract Mock - - - - -
+
+    const createTokenExpectedResult = await htsSystemContract.connect(deployer).callStatic.createFungibleToken(token, initialTotalSupply, decimals);
+    const expectedTokenAddress = createTokenExpectedResult.tokenAddress; // TODO: investigate why computing the expected address using the factory address and nonce doesn't work
```
Comment (AlfredoG87) [inline tools/hardhat-example/test/simple-vault.js:91]: Is this TODO still pending?

**Review by AlfredoG87 — CHANGES_REQUESTED** [review]
LG, just 1 nit and a question

**Comment by AlfredoG87** [comment]
Also, in order for the tests to pass, please rebase to latest version of main.

File: tools/hardhat-example/test/simple-vault.js (hunk 2 above)
Comment (mshakeg) [reply tools/hardhat-example/test/simple-vault.js:91]: fixed

**Comment by mshakeg** [comment]
Hey @Nana-EC do you think there'd be significant value in say applying these changes to an existing hardhat project such as [Uniswap/v3-core](https://github.com/Uniswap/v3-core) to demonstrate how projects can easily adapt their codebases for Hedera? If so, I might consider doing it.

**Review by AlfredoG87 — COMMENTED** [review]
Looks good, please rebase and resolve conflicts.

**Comment by AlfredoG87** [comment]
it seems like the `package.json` and `package-lock.json` are out of sync after the rebase, could you just run 
`npm install` and push the changes to the `package-lock.json` file? This is making the tests fail

**Review by AlfredoG87 — CHANGES_REQUESTED** [review]
please update `package-lock.json` by running `npm install`

**Comment by Nana-EC** [comment]
> Hey @Nana-EC do you think there'd be significant value in say applying these changes to an existing hardhat project such as [Uniswap/v3-core](https://github.com/Uniswap/v3-core) to demonstrate how projects can easily adapt their codebases for Hedera? If so, I might consider doing it.

@mshakeg great idea, I think adding a `contracts/uniswap/v3-core` example under https://github.com/hashgraph/hedera-smart-contracts/tree/main/contracts which matching test support would be the right approach.
Did you want to create an issue or PR for this in that repo?

**Comment by mshakeg** [comment]
@Nana-EC no I did not, I think it would be better to fork the uniswap/v3-core repo, apply the changes and perhaps leave a PR open from the fork to `uniswap/v3-core@main` for developers to easily see the changes necessary for integrating with Hedera.

**Comment by mshakeg** [comment]
> please update `package-lock.json` by running `npm install`

**Comment by mshakeg** [comment]
@AlfredoG87 hopefully that's the last time I have to do that :)

**Comment by AlfredoG87** [comment]
@mshakeg Thank you!
Most of the tests are passing now, there is only an issue with the `dapp-tests`

```Error HH18: You installed Hardhat with a corrupted lockfile due to the NPM bug #4828.

Please delete your node_modules, package-lock.json, reinstall your project, and try again.

HardhatError: HH18: You installed Hardhat with a corrupted lockfile due to the NPM bug #4828.
```
see [here](https://github.com/hashgraph/hedera-json-rpc-relay/actions/runs/7363125172/job/20107602135?pr=1884)

Please follow the instructions and push the changes to the `package-lock.json` file after is fixed.

**Comment by mshakeg** [comment]
@AlfredoG87 I've merged main into this branch, so those files(`package.json` and `package-lock.json`) are now unchanged.

**Comment by mshakeg** [comment]
@AlfredoG87 seems like it's in a much better state. It failed initially due to me having installed with node 20, but the workflow is currently set to use node 18, so I re-installed using that version, and it then failed due to a compilation error(seems like there's been changes to the mocks I initially developed?) so I locked it to use my version of the mocks.

**Review by AlfredoG87 — DISMISSED** [review]

**Comment by AlfredoG87** [comment]
@mshakeg 

I am sorry, there are some conflicts on the `package.json` and `package-lock.json` again 😓 
I should have merged it when I had the chance, just wanted to give folks some time to review in case there were other comments.

once this rebase and conflicts are resolved I will re-approve and merge.

**Comment by AlfredoG87** [comment]
@mshakeg 
we are seeing the error 
```
Please delete your node_modules, package-lock.json, reinstall your project, and try again.
```
https://github.com/hashgraph/hedera-json-rpc-relay/actions/runs/7477530404/job/20350428264?pr=1884

is it the node version again?

**Comment by mshakeg** [comment]
@AlfredoG87 seems so. Hopefully that fixes it.

**Review by AlfredoG87 — APPROVED** [review]
LGTM

**Pull request merged**